### ISSUE TITLE
Proposal: Add getAsCollection() method

### ISSUE THREAD

**Issue by jodiedunlop** [rexlabsio/array-object-php#8]
When you call `get('some.node')` if the node doesn't exist you get `null` (or pass a default value) .

Suggest adding `getAsCollection()` to ensure a collection is _always_ returned regardless of whether the node exists, or if the node is an item/collection. This is useful due to chaining.

Example:
```
$response->getAsCollection('recipients.data')->each(function($recipient) { ... }); 
```
- If the node doesn't exist an empty (ArrayObject) collection will be returned
- If the node exists but isn't a collection (eg. a scalar or an associative array), then it will be wrapped in a collection with a single item

**Comment by cerbero90** [comment]
Nice proposal, Alex went through this a couple of days ago if I'm not wrong.

It's just a matter of personal taste but what do you think about:
- `$response->collect('recipients.data')` instead of `getAsCollection()`
- `$response->first('recipients.data')` instead of `getAsItem()`

Perhaps `getAsCollection()` and `getAsItem()` are more descriptive though.

**Comment by jodiedunlop** [comment]
I like `collect()` and `first()` already exists but it does not ensure you get a single item (eg. the first item in a collection could be a collection)

**Comment by jodiedunlop** [comment]
I mean you can also do `$response->get('recipients.data.0')`  so maybe it's not needed at all, just the `collect('node')` method

**Comment by jodiedunlop** [comment]
Yeah the problem with `xxitem()` is you can't ensure they get an item.  You either get a scalar value for primitive types or a boxed item.  Boxing scalars would be a really bad idea.

I think you were right with `first()` and i was overthinking it.
